### ISSUE TITLE
bertsquad-10 and 12 NNPA performance regression

### ISSUE THREAD

**Issue by cjvolzka** [onnx/onnx-mlir#2633]
Starting with commit 5db6e46, the [bertsquad-10](https://github.com/onnx/models/blob/main/text/machine_comprehension/bert-squad/model/bertsquad-10.onnx) and [bertsquad-12](https://github.com/onnx/models/blob/main/text/machine_comprehension/bert-squad/model/bertsquad-12.onnx) modes have a 10x performance reduction compared to the previous commit 5f46d52. 

Focusing on bertsquad-12 using the example input from the onnx model zoo, in 5f46d52 an inference takes 103.609 ms but In 5db6e46 the same input takes 1749.15 ms.

The output of --onnx-op-stats shows several changes but I'm guessing the most relevant will be the 24 MatMuls that go to CPU instead of NNPA. 

commit 5f46d52 (103.609 ms at runtime)
```
--EmitLib --mtriple=s390x-ibm-loz --mcpu=z16 --maccel=NNPA ./bertsquad-12.onnx --onnx-op-stats TXT
Operations encountered:
-----------------------
   func.func               , 1
   func.return             , 1
   onnx.Add                , 63
   onnx.Cast               , 105
   onnx.Concat             , 91
   onnx.Constant           , 78
   onnx.Dim                , 41
   onnx.Expand             , 1
   onnx.Gather             , 1
   onnx.Max                , 12
   onnx.Mul                , 149
   onnx.NoValue            , 1
   onnx.OneHot             , 1
   onnx.Reciprocal         , 25
   onnx.ReduceMeanV13      , 50
   onnx.Reshape            , 166
   onnx.Split              , 1
   onnx.Sqrt               , 25
   onnx.Squeeze            , 4
   onnx.Sub                , 62
   onnx.Transpose          , 49
   onnx.Unsqueeze          , 13
  zhigh.Add                , 49
  zhigh.MatMul             , 98
  zhigh.Mul                , 61
  zhigh.Softmax            , 12
  zhigh.Stick              , 186
  zhigh.StickifiedConstant , 147
  zhigh.Tanh               , 12
  zhigh.Unstick            , 159

```

5db6e46 (1749.15 ms at runtime)
```
--EmitLib --mtriple=s390x-ibm-loz --mcpu=z16 --maccel=NNPA ./bertsquad-12.onnx --onnx-op-stats TXT
Operations encountered:
-----------------------
   func.func               , 1
   func.return             , 1
   onnx.Add                , 63
   onnx.Cast               , 109
   onnx.Concat             , 72
   onnx.Constant           , 79
   onnx.Dim                , 33
   onnx.Expand             , 1
   onnx.Gather             , 1
   onnx.MatMul             , 24
   onnx.Max                , 12
   onnx.Mul                , 149
   onnx.NoValue            , 1
   onnx.OneHot             , 1
   onnx.Reciprocal         , 25
   onnx.ReduceMeanV13      , 50
   onnx.Reshape            , 94
   onnx.Slice              , 5
   onnx.Split              , 1
   onnx.Sqrt               , 25
   onnx.Squeeze            , 4
   onnx.Sub                , 62
   onnx.Transpose          , 49
   onnx.Unsqueeze          , 13
  zhigh.Add                , 49
  zhigh.MatMul             , 74
  zhigh.Mul                , 61
  zhigh.Softmax            , 12
  zhigh.Stick              , 138
  zhigh.StickifiedConstant , 147
  zhigh.Tanh               , 12
  zhigh.Unstick            , 135
```

**Comment by cjvolzka** [comment]
I confirmed today the runtimes are still about 10x worse than the 0.4.1 release.

@tungld could you take a look since it appeared following the https://github.com/onnx/onnx-mlir/commit/5db6e462abbe746e3a138b08643d0bd57f4a516c commit?

**Comment by tungld** [comment]
@cjvolzka sure, I am looking at these models.

**Comment by tungld** [comment]
I created a PR to fix the regression: #2724.
